scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskDBUtils.java

```diff
@@ -27,6 +27,9 @@
                 Criteria criteria = session.createCriteria(TaskData.class);
                 List<TaskState> result = null;
                 
+                // only fetch non-removed tasks
+                criteria.createAlias("jobData", "job").add(Restrictions.eq("job.removedTime", -1L));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "jobData" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)